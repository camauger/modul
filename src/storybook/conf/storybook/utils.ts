export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';

