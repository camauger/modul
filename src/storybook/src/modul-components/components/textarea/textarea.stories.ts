import { actions } from '@storybook/addon-actions';
import { TEXTAREA_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${TEXTAREA_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    data: () => ({
        model: ''
    }),
    methods: actions(
        'input',
        'focus',
        'blur'
    ),
    template: `<div>
        <m-textarea v-model="model"
                    @input="input"
                    @focus="focus"
                    @blur="blur"></m-textarea>
        <br/>model value = {{model}}
    </div>`
});

defaultStory.story = {
    name: 'default'
};

export const state = () => `<div>
    <m-textarea label="Disabled"
                value="Lorem Ipsum"
                :disabled="true"></m-textarea>
    <m-textarea label="Readonly"
                value="Lorem Ipsum"
                :readonly="true"></m-textarea>
    <m-textarea label="Waiting"
                value="Lorem Ipsum"
                :waiting="true"></m-textarea><br>
    <m-textarea label="Disabled and error"
                value="Lorem Ipsum"
                :disabled="true"
                error-message="Error"></m-textarea>
    <m-textarea label="Readonly and error"
                value="Lorem Ipsum"
                :readonly="true"
                error-message="Error"></m-textarea>
    <m-textarea label="Waiting and error"
                value="Lorem Ipsum"
                :waiting="true"
                error-message="Error"></m-textarea><br>
    <m-textarea label="Disabled and readonly"
                value="Lorem Ipsum"
                :disabled="true"
                :readonly="true"></m-textarea>
    <m-textarea label="Disabled and waiting"
                value="Lorem Ipsum"
                :disabled="true"
                :waiting="true"></m-textarea>
    <m-textarea label="Readonly and waiting"
                value="Lorem Ipsum"
                :readonly="true"
                :waiting="true"></m-textarea>
    <m-textarea label="Disabled, readonly and waiting"
                value="Lorem Ipsum"
                :disabled="true"
                :readonly="true"
                :waiting="true"></m-textarea>
</div>`;
