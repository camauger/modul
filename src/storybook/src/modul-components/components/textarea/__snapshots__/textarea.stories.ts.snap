// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-textarea State 1`] = `
<div>
  <div
    class="m-textarea m--is-disabled m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-label-up m--is-disabled m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Disabled
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                disabled="disabled"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-textarea m--is-readonly m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-readonly m--is-label-up m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Readonly
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                readonly="readonly"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-textarea m--is-waiting m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-label-up m--is-waiting m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Waiting
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <div
                class="m-spinner m-input-style__spinner"
              >
                <!---->
                 
                <div />
              </div>
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
  <br />
   
  <div
    class="m-textarea m--is-disabled m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-label-up m--is-disabled m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Disabled and error
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                disabled="disabled"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-textarea m--is-readonly m--has-error m--has-label m--has-validation-message"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-readonly m--is-label-up m--has-error m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Readonly and error
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                readonly="readonly"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <p
          class="m-validation-message__error"
        >
          <svg
            class="m-icon m-validation-message__icon"
            height="1em"
            width="1em"
          >
            <title>
              Erreur
            </title>
             
            <use
              aria-hidden="true"
              class=""
            />
          </svg>
           
          <span
            class="m-validation-message__text"
          >
            Error
          </span>
        </p>
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-textarea m--is-waiting m--has-error m--has-label m--has-validation-message"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-label-up m--is-waiting m--has-error m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Waiting and error
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <div
                class="m-spinner m-input-style__spinner"
              >
                <!---->
                 
                <div />
              </div>
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <p
          class="m-validation-message__error"
        >
          <svg
            class="m-icon m-validation-message__icon"
            height="1em"
            width="1em"
          >
            <title>
              Erreur
            </title>
             
            <use
              aria-hidden="true"
              class=""
            />
          </svg>
           
          <span
            class="m-validation-message__text"
          >
            Error
          </span>
        </p>
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
  <br />
   
  <div
    class="m-textarea m--is-disabled m--is-readonly m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-readonly m--is-label-up m--is-disabled m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Disabled and readonly
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                disabled="disabled"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                readonly="readonly"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-textarea m--is-disabled m--is-waiting m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-label-up m--is-disabled m--is-waiting m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Disabled and waiting
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                disabled="disabled"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <div
                class="m-spinner m-input-style__spinner"
              >
                <!---->
                 
                <div />
              </div>
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-textarea m--is-waiting m--is-readonly m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-readonly m--is-label-up m--is-waiting m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Readonly and waiting
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                readonly="readonly"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <div
                class="m-spinner m-input-style__spinner"
              >
                <!---->
                 
                <div />
              </div>
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <div
    class="m-textarea m--is-disabled m--is-waiting m--is-readonly m--has-label"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-readonly m--is-label-up m--is-disabled m--is-waiting m--has-label m--has-value m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <label
            class="m-input-style__label"
            for="mTextarea-fakeId"
            style="margin-right: 0px;"
          >
            <span
              class="m-input-style__text"
            >
              <span>
                Disabled, readonly and waiting
              </span>
               
              <!---->
            </span>
          </label>
           
          <div
            class="m-input-style__input"
          >
            <!---->
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                disabled="disabled"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                readonly="readonly"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <div
                class="m-spinner m-input-style__spinner"
              >
                <!---->
                 
                <div />
              </div>
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-textarea default 1`] = `
<div>
  <div
    class="m-textarea"
    style="width: 100%; max-width: 288px;"
  >
    <div
      class="m-input-style m--is-tag-default"
      style="margin-top: 10px;"
    >
      <div
        class="m-input-style__main"
      >
        <div
          class="m-input-style__body"
        >
          <span
            class="m-input-style__label"
            style="z-index: -1;"
          >
            <span
              class="m-input-style__text"
            />
          </span>
           
          <div
            class="m-input-style__input"
          >
            <div
              class="m-input-style__prefix"
            />
             
            <div
              class="m-input-style__content"
            >
              <textarea
                class="m-textarea__input"
                id="mTextarea-fakeId"
                maxlength="Infinity"
                rows="1"
              />
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="m-input-style__suffix"
            >
               
              <!---->
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="m-textarea__validation"
    >
      <div
        class="m-validation-message m-textarea__validation__message"
      >
        <!---->
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="resize-sensor"
      style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
    >
      <div
        class="resize-sensor-expand"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
        />
      </div>
      <div
        class="resize-sensor-shrink"
        style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%"
        />
      </div>
    </div>
  </div>
   
  <br />
  model value = 
    
</div>
`;
